game-core/src/main/java/games/strategy/triplea/delegate/battle/steps/fire/FiringGroup.java

```diff
@@ -0,0 +1,125 @@
+package games.strategy.triplea.delegate.battle.steps.fire;
+
+import com.google.common.collect.ArrayListMultimap;
+import com.google.common.collect.Multimap;
+import games.strategy.engine.data.Unit;
+import games.strategy.engine.data.UnitType;
+import games.strategy.triplea.delegate.Matches;
+import java.util.Collection;
+import java.util.Collections;
+import java.util.List;
+import java.util.stream.Collectors;
+import lombok.Value;
+import org.triplea.java.collections.CollectionUtils;
+
+/**
+ * Group of units that are firing on targets
+ *
+ * <p>If the group is suicideOnHit, then all of the units should have the same unit type
+ */
+@Value
+public class FiringGroup {
+
+  private static final UnitType NON_SUICIDE_MULTIMAP_KEY = new UnitType("nonsuicide", null);
+
+  String displayName;
+  String groupName;
+  Collection<Unit> firingUnits;
+  Collection<Unit> targetUnits;
+  boolean suicideOnHit;
+
+  private FiringGroup(
+      final String displayName,
+      final String groupName,
+      final Collection<Unit> firingUnits,
+      final Collection<Unit> targetUnits) {
+    this.displayName = displayName;
+    this.groupName = groupName;
+    this.firingUnits = firingUnits;
+    this.targetUnits = targetUnits;
+    this.suicideOnHit = this.firingUnits.stream().allMatch(Matches.unitIsSuicideOnHit());
+  }
+
+  public Collection<Unit> getTargetUnits() {
+    return Collections.unmodifiableCollection(targetUnits);
+  }
+
+  public Collection<Unit> getFiringUnits() {
+    return Collections.unmodifiableCollection(firingUnits);
+  }
+
+  /**
+   * Keeps alive units around
+   *
+   * <p>Units unfortunately don't track their own status so the list of targets needs to be updated
```

Comment:
I think this comment is more of an implementation note. Ideally this class is designed so that it is generic and has a role, a function, and is not aware of limitations around it. IF this class is responsible for tracking units that are still alive, perhaps that can just be cleanly documented in the class javadoc? AS-is, it's a bit odd to see this note, I'm not sure if it really answers questions, seems like if you have context then it makes more sense but otherwise is a bit confusing.